### PR TITLE
Fix `editable?` function so that it respects native write permissions

### LINKED ISSUE #37765: [MLv2][BE] `editable?` function doesn't take into account native write permissions
When working on https://github.com/metabase/metabase/pull/37744, I realized that `editable?` function doesn't take into account native write permissions.
https://github.com/metabase/metabase/blob/1a411da8a27094bfbf6acdfd23e2ed70c7d1037a/src/metabase/lib/metadata.cljc#L213

It currently returns `true` for native queries even if the database doesn't have `native_permissions` set to `write`.
https://github.com/metabase/metabase/blob/1a411da8a27094bfbf6acdfd23e2ed70c7d1037a/src/metabase/api/database.clj#L80

This check has already been implemented in https://github.com/metabase/metabase/blob/backport-1a411da8a27094bfbf6acdfd23e2ed70c7d1037a/src/metabase/lib/native.cljc#L227

I tried to solve the problem in [this commit](https://github.com/metabase/metabase/pull/37744/commits/1175300250f5767a82da95b323a3e6b532bb8e79) but ran into circular dependency problems. Thought I had a workaround, but tests are still failing.

### PR DESCRIPTION
**Pull request by nemanjaglumac**
Fixes #37765 and unblocks https://github.com/metabase/metabase/issues/37389

This PR adjusts `editable?` function that now takes native permissions into account.
It also adds a few more test cases where those native permissions are explicitly set to `:none` or are missing altogether.

> [!important]
> This PR alters the test database metadata - it adds `:native-permissions :write` to the map. In other words, the default database assumes native write permissions. This wasn't explicitly set previously which was the equivalent of no native permissions.

### CODE REVIEW

**Review by metamben — APPROVED** [review]
I'm in fact surprised that native permissions play a role in deciding if a question can be edited. I might be misunderstanding what `:native-permissions` mean.

File: src/metabase/lib/metadata.cljc [hunk 1]
```diff
@@ -224,4 +225,8 @@
                     (= (:database query) id))
                   (or (and source-table (table query source-table))
                       (and source-card  (card  query source-card))
-                      (= (:lib/type stage0) :mbql.stage/native))))))
+                      (and
+                        (= (:lib/type stage0) :mbql.stage/native)
+                        ;; Couldn't import and use `lib.native/has-write-permissions` here due to a circular dependency
+                        ;; TODO Find a way to unify has-write-permissions and this function?
+                        (= :write (:native-permissions (database query)))))))))
```
[the remark this answers is not in the document]
Comment (nemanjaglumac) [reply src/metabase/lib/metadata.cljc:232]: I believe so.
https://github.com/metabase/metabase/blob/b159e818a428b01008a2792dea619e21b100cc8d/test/metabase/lib/native_test.cljc#L282:L286

**Comment by nemanjaglumac** [comment]
> I'm in fact surprised that native permissions play a role in deciding if a question can be edited. I might be misunderstanding what :native-permissions mean.

They reflect the choice set in admin > permissions > native query editing.
image

yes -> :write
no -> :none
